### ISSUE TITLE
Hardware detect misidentifies AMD Strix Halo as ULTRA LOW

### ISSUE THREAD

**Issue by ejstembler** [Pavelevich/llm-checker#112]
Running `npx llm-checker check` outputs:

```console
SYSTEM SUMMARY
──────────────────────────────────────────────────
CPU: RYZEN AI MAX+ 395 w/ Radeon 8060S (32 cores)
Memory: 124GB RAM
GPU: AMD GPU (1586)
Architecture: x64
Backend: AMD ROCm
Hardware Tier: ULTRA LOW
```

Or running `npx llm-checker hw-detect` outputs:

```console
llm-checker | Hardware Detection
------------------------------------------
✔ Hardware detected!

=== Hardware Detection ===

Summary:
  AMD GPU (1586) + [AMD/ATI] Device 1586 (1GB VRAM) + AMD RYZEN AI MAX+ 395 w/ Radeon 8060S
  Tier: ULTRA LOW
  Max model size: 0GB
  Best backend: AMD ROCm
  Dedicated GPUs: AMD GPU (1586) + [AMD/ATI] Device 1586
  Integrated GPUs: None

CPU:
  AMD RYZEN AI MAX+ 395 w/ Radeon 8060S
  Cores: 32 (16 physical)
  SIMD: AVX512
  [OK] AVX-512
  [OK] AVX2

ROCM:
  ROCm: null
  Total VRAM: 1GB
  AMD GPU (1586): 1GB

GENERIC:
  Source: systeminformation
  Total dedicated VRAM: 1GB
  [AMD/ATI] Device 1586: 1GB (Discrete)

Fingerprint: rocm--gpu-(1586)-1gb
```

Actual hardware specs:

- AMD Ryzen AI MAX+395
- AMD Radeon 8060S iGPU
- 128GB LPDDR5X 8000 

From Gemini:

`llm-checker` rated your system as **ULTRA LOW** because of a classic hardware-detection fallback issue.

Your machine (an AMD Ryzen AI MAX+ 395 "Strix Halo" APU with integrated Radeon 8060S graphics) is actually a **high-end local AI monster**, but `llm-checker` misidentified it.

---

### Why It Rated You "ULTRA LOW"

1. **Unrecognized GPU Device ID (`AMD GPU (1586)`):**
`llm-checker` maps GPU strings to hardcoded VRAM tables (e.g., matching strings like `RTX 4090` or `RX 7900 XTX`). Because the Radeon 8060S is an integrated GPU on the new Strix Halo platform, the system returned a raw PCI ID (`1586`). The tool failed to look up its actual specs and defaulted to **0 MB VRAM**.
2. **Offloading Logic Defaults to VRAM over RAM:**
`llm-checker` evaluates discrete VRAM first. When it detects zero (or negligible) dedicated VRAM on a Windows/Linux machine, it categorizes the system as a "very limited system" meant only for `<3B models` (the definition of the **ULTRA LOW** tier).
3. **Unified Memory Ignored:**
Unlike Apple Silicon (which `llm-checker` explicitly handles as unified memory), AMD's APU unified architecture requires checking Linux GTT or Windows Variable Graphics Memory (VGM) allocations. The tool treated your system like an older, low-end integrated Intel/AMD HD Graphics chip.

---

### What Your Machine *Actually* Is

Your Ryzen AI MAX+ 395 machine is one of the most capable consumer local-LLM setups available:

* **Unified Pool:** With **124 GB RAM**, you can assign up to ~96 GB–110 GB directly as VRAM to your ROCm/Radeon backend.
* **True Tier:** **FLAGSHIP / ULTRA HIGH**.
* **What You Can Actually Run:** You can run massive models that dedicated $2,000 GPUs cannot load due to VRAM limits — including **70B parameter models (Llama 3.3 70B)**, **32B models at high precision**, and heavy **Mixture-of-Experts (Qwen 235B / DeepSeek R1 MoE quantized)**.

**Comment by signerless** [comment]
Fixed in #113 and merged to `main` (`9f49eda`).

Root cause confirmed: device `1002:1586` appeared once through ROCm and again through `systeminformation`. The generic view had no separate `deviceId`, was classified as a 1 GB discrete GPU, and prevented the unified/shared-memory path from being used.

The detector now:
- recognizes `1586` as the integrated AMD Strix Halo family
- extracts the PCI id from the model string when `deviceId` is absent
- reconciles the ROCm and generic views as one physical iGPU
- uses the shared/unified pool for model-fit calculations instead of the 1 GB aperture
- preserves `8050S` or `8060S` when a source reports that SKU; the PCI id alone stays family-neutral because both SKUs share it

With the reproduced 124 GB / 1 GB-aperture fixture, the result is one integrated Strix Halo GPU, 118 GB effective memory, a model envelope above 100 GB, and no `ULTRA LOW` classification.

Validation before merge:
- exact ROCm + `[AMD/ATI] Device 1586` flow without `deviceId`
- 8050S and 8060S name-preservation regressions
- AMD, hardware-tier, detector, and AI Check regressions
- full suite: 55/55
- GitHub CI: Node 18/20/22, Windows smoke, and policy gate all passed

Thanks for including both command outputs and the raw device id; that made the duplicate-inventory failure reproducible.
